docker-compose.yaml

```diff
@@ -74,6 +74,69 @@ services:
       - elasticsearch
     command: postgres:5432 mysql:3306 minio:9000 inbucket:10080 openldap:389 elasticsearch:9200
 
+  leader:
+    build:
+      context: .
+      dockerfile: ./build/Dockerfile.buildenv
+    working_dir: '/home/mattermost-server'
+    environment:
+      - "MM_SQLSETTINGS_DATASOURCE=postgres://mmuser:mostest@postgres/mattermost_test?sslmode=disable\u0026connect_timeout=10"
+    networks:
+      - mm-test
+    depends_on:
+      - start_dependencies
+    volumes:
+      - './:/home/mattermost-server'
+      - './../mattermost-webapp:/home/mattermost-webapp'
+      - './../enterprise:/home/enterprise'
+    restart: on-failure
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://leader:8065/api/v4/system/ping"]
+      interval: 5s
+      timeout: 10s
+      retries: 30
+    command: ['make', 'run-server-no-docker']
+    expose:
+      - "8065"
+
+  follower:
```

Comment:
This is identical to the `leader` service, but cannot use `--scale service=2` because upon initialization those two will compete for migrations.

Reply:
Hmm .. yeah good point. We should work on adding a DB lock and only then move towards making schema changes. Oh well, for another time I guess!